### ISSUE TITLE
the space of the font "新宋体“ doesn't display properly

### ISSUE THREAD

**Issue by HuangDuoYan** [microsoft/vscode#188774]
Type: <b>Bug</b>

I changed the font in the editor to "新宋体", then found that half corner Spaces appear the same as full corner spaces. But the font is the default in Chinese in Visual Studio and the space displays normally.

VS Code version: Code 1.80.1 (74f6148eb9ea00507ec113ec51c489d6ffb4b771, 2023-07-12T17:22:07.651Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Remote OS version: Linux x64 5.15.0-78-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i7-1260P (16 x 2496)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.73GB (5.32GB free)|
|Process Argv|--crash-reporter-id 32ce9b42-12bd-41b0-86e8-d8a740da346d|
|Screen Reader|no|
|VM|50%|

|Item|Value|
|---|---|
|Remote|SSH: ubuntu|
|OS|Linux x64 5.15.0-78-generic|
|CPUs|12th Gen Intel(R) Core(TM) i7-1260P (2 x 2496)|
|Memory (System)|3.79GB (1.41GB free)|
|VM|100%|
</details><details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
deveco-device-tool-remote|Hua|3.1.600
remote-ssh|ms-|0.102.0
remote-ssh-edit|ms-|0.86.0
remote-explorer|ms-|0.4.1
vscode-gn|npc|1.1.0
deveco-chip-config|Hua|3.1.600
deveco-device-tool|Hua|3.1.600
deveco-device-tool-debug|Hua|3.1.600
vscode-language-pack-zh-hans|MS-|1.80.2023071209
cmake-tools|ms-|1.14.34
cpptools|ms-|1.16.3
cpptools-extension-pack|ms-|1.3.0
cmake|twx|0.0.17

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
2e4cg342:30602488
89544117:30613380
a9j8j154:30646983
showlangstatbar:30737416
vsctsb:30748421
pythonfmttext:30731395
pythoncmv:30756943
fixshowwlkth:30771522
showindicator:30785052
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxpt:30784022
pythonnoceb:30797650
e537b577:30795824
dsvsc013:30795093
dsvsc014cf:30797590
cmakestatusbarv2:30779594

```

</details>

**Comment by buger404** [comment]
Maybe using the term `NSimsun` instead of `新宋体` would make it easier for developers to deal with this issue.

**Comment by rzhao271** [comment]
Yeah, it looks like all spaces are full-width spaces, even though other half-width characters are rendered correctly.

**Comment by dddddgz** [comment]
我也有这个问题，只好用仿宋了……主要是中英文不能对齐

**Comment by xhawk18** [comment]
修好了吗？这个问题貌似还在。
喜欢宋体没有模糊边界的硬朗感。

**Comment by xhawk18** [comment]
[xspace.zip](https://github.com/microsoft/vscode/files/14035239/xspace.zip)
![1](https://github.com/microsoft/vscode/assets/4671118/dae93fb7-1b05-4e1c-be2e-6fa0c9d46064)
![2](https://github.com/microsoft/vscode/assets/4671118/43be763a-1d94-405e-8572-98bcb0f6f17d)

As a workaround, I just create a font file named "xspace" with only one space characters.
Install the font, and change the font settings to "xspace,NSimSun" should be OK.

**Comment by dddddgz** [comment]
That's a good idea! Create an normal-size space instead

用正常大小的空格替换确实是好方法
